src/Gui/MainWindowPy.cpp

```diff
@@ -215,3 +224,57 @@ Py::Object MainWindowPy::removeWindow(const Py::Tuple& args)
     }
     return Py::None();
 }
+
+Py::Object MainWindowPy::showHint(const Py::Tuple& args)
```

Comment:
Could this in any way be used to run arbitrary code? I didn't look it through, but just my thought, when I see string interpolation and user input in the same PR.

Reply:
The hint is not stored anywhere, this is just function exposed to python. Given that python is arbitrary code execution packaged as feature this is not an issue.